ev3dev/display.py

```diff
@@ -267,3 +277,77 @@ def update(self):
             self.mmap[:] = self._img_to_rgb565_bytes()
         else:
             raise Exception("Not supported")
+
+    def image_filename(self, filename, clear_screen=True, x1=0, y1=0, x2=None, y2=None):
+
+        if clear_screen:
+            self.clear()
+
+        filename_im = Image.open(filename)
+
+        if x2 is not None and y2 is not None:
+            return self._img.paste(filename_im, (x1, y1, x2, y2))
+        else:
+            return self._img.paste(filename_im, (x1, y1))
+
+    def shapes_line(self, clear_screen=True, x1=0, y1=0, x2=0, y2=0, line_color='black', width=1):
+
+        if clear_screen:
+            self.clear()
+
+        return self.draw.line((x1, y1, x2, y2), fill=line_color, width=width)
+
+    def shapes_circle(self, clear_screen=True, x=50, y=50, radius=40, fill_color='black', outline_color='black'):
+
+        if clear_screen:
+            self.clear()
+
+        x1 = x - radius
+        y1 = y - radius
+        x2 = x + radius
+        y2 = y + radius
+
+        return self.draw.ellipse((x1, y1, x2, y2), fill=fill_color, outline=outline_color)
```

Comment:
What would happen if you passed, say, `red` to this function in the EV3 (the EV3 has a monochrome display)? Does it convert the color brightness to a single greyscale channel, or do something else?

Reply:
`red` it converted to `black` on my EV3. I don't know if there is a cutoff where something close to white translates to white instead of black or not.